resources/qml/pages/MessagingPage.qml

```diff
@@ -147,12 +191,38 @@ Page {
                     }
                     text: resource;
                 }
-
                 Row {
                     spacing: 5
                     anchors.right: (!item.alignRight ? parent.right : undefined)
+
+                    Label {
+                        id: upload
+                        visible: msgstate == 4
+                        color: Theme.secondaryColor;
+                        font {
+                            family: Theme.fontFamilyHeading;
+                            pixelSize: Theme.fontSizeTiny;
+                        }
+
+                        Connections {
+                            target: shmoose
+                            onSignalShowStatus: {
+                                if(headline == "File Upload" && msgstate == 4)
```

Comment:
"File Upload" needs to be wrapped in tr(). Also the text in the emitting signal. Otherwise it will break if someone fixes the message translation in the signal.

Reply:
Agreed.

Reply:
Maybe its even more clever to use a separate signal to transport the upload status. If someone changes the headline by just one character, the functionality is broken. And no one would expect functionality to be bound to a translation text.